name: Sanity Check

on:
  pull_request_target:
    types: [opened, synchronize, reopened, ready_for_review]

jobs:
  valid-user:
    runs-on: ubuntu-20.04
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Sanity Check PR Content
        id: sanity_check_pr_content
        continue-on-error: true
        run: |
          python3 -m venv ve1
          cd scripts && ../ve1/bin/pip3 install -r requirements.txt && cd ..
          cd scripts && ../ve1/bin/python3 setup.py install && cd ..
          ./ve1/bin/sanity-check-pr --pr-number=${{ github.event.number }}

      - uses: actions/github-script@v3
        if: ${{ steps.sanity_check_pr_content.conclusion == 'success'}}
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            github.issues.addLabels({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              labels: ['ok-to-test']
            })

      - uses: actions/github-script@v3
        if: ${{ steps.sanity_check_pr_content.conclusion == 'failure'}}
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            github.issues.removeLabel({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              name: 'ok-to-test'
            })
